Docs/Book/GettingStartedInC++.md

```diff
@@ -0,0 +1,1420 @@
+as gibberish if printed to a screen. The RDKit pickle format is fairly
+compact and it is much, much faster to build a molecule from a pickle
+than from a Mol file or SMILES string, so storing as pickles molecules
+you will be working with repeatedly can be a good idea:
+[(example12.cpp)](./C++Examples/example12.cpp):
+```c++
+// writing to pickle file
+std::string smi_file = getenv("RDBASE");
+smi_file += "/Code/GraphMol/test_data/canonSmiles.long.smi";
+std::string pkl_name = "canonSmiles.long.pkl";
+
+// tab-delimited file, SMILES in column 0, name in 1, no title line
+RDKit::SmilesMolSupplier suppl( smi_file , "\t" , 0 , 1 , false );
+std::ofstream pickle_ostream( pkl_name , std::ios_base::binary );
+int write_cnt = 0;
+while( !suppl.atEnd() ) {
+  RDKit::ROMol *mol = suppl.next();
+  RDKit::MolPickler::pickleMol( *mol , pickle_ostream );
+  delete mol;
+  ++write_cnt;
+}
+pickle_ostream.close();  
+std::cout << "Wrote " << write_cnt << " molecules" << std::endl;
+
+// reading from pickle file
+std::ifstream pickle_istream( pkl_name , std::ios_base::binary );
+int read_cnt = 0;
+while( !pickle_istream.eof() ) {
+  RDKit::ROMol mol3;
+  try {
+    RDKit::MolPickler::molFromPickle( pickle_istream , mol3 );
+  } catch( RDKit::MolPicklerException &e ) {
+    break;
+  }
+  ++read_cnt;
+}
+pickle_istream.close();
+std::cout << "Read " << read_cnt << " molecules." << std::endl;
+```
+However, currently the pickling process does not preserve and
+properties attached to the molecule, which included the molecule name
+(property "_Name"). This is likely to change in 2017.
+
+### Drawing Molecules
+The RDKit has some built-in functionality for drawing molecules, found
+in the RDKit namespace, with header files in
+`$RDBASE/Code/GraphMol/MolDraw2D`.  There is an abstract base class
+MolDraw2D which defines the interface and does the drawing, with
+concrete classes for drawing to SVG or PNG files and Qt and wx
+widgets.  Only the SVG output is built by default, Cairo support
+requires the argument `-DRDK_BUILD_CAIRO_SUPPORT=ON` to cmake, and Qt
+support `-DRDK_BUILD_QT_SUPPORT=ON`. To create an SVG file:
+[(example13.cpp)](./C++Examples/example13.cpp):
+```c++
+#include <GraphMol/MolDraw2D/MolDraw2DSVG.h>
+.
+.
+RDKit::SDMolSupplier mol_supplier( "data/cdk2.sdf" , true );
+RDKit::ROMol *mol1 = mol_supplier.next();
+RDDepict::compute2DCoords( *mol1 );
+std::ofstream outs("cdk_mol1.svg");
+RDKit::MolDraw2DSVG svg_drawer(300, 300, outs);
+svg_drawer.drawMolecule( *mol1 );
+svg_drawer.finishDrawing();
+outs.close();
+```
+The procedure for a PNG is slightly different:
+[(example13.cpp)](./C++Examples/example13.cpp):
+```c++
+#include <GraphMol/MolDraw2D/MolDraw2DSVG.h>
+.
+.
+RDKit::MolDraw2DCairo cairo_drawer(300, 300);
+cairo_drawer.drawMolecule(*mol1);
+cairo_drawer.finishDrawing();
+cairo_drawer.writeDrawingText("cdk_mol1.png");
+```
+The Python wrapper includes the function `Chem.MolsToGridImage`.
+There is, as yet, no equivalent in the C++ version.
```

Comment:
there is! Here's sample code: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/MolDraw2D/test1.cpp#L1567